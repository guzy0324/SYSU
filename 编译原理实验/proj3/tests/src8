/* EXAMPLE 4: Two classes this time */
program x4;
class Point {
	     declarations
        		int x, y;
	     enddeclarations
}

class Test {
	declarations
            	Point p0;
	enddeclarations
        method void main()
	     declarations
                	Point p;
		        int xxx;
	     enddeclarations
	{
	        p.y := 1;
	        p.x := 1;
        }
}

