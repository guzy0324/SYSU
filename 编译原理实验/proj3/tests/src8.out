********************************Symbol Table************************************

          Name Nest-Level  Tree-Node Predefined        Kind      Type      Value     Offset  Dimension     Argnum

  1      Point          0                            class                                                       
  2          y          1                         variable  type-tree                                            
  3          x          1                         variable  type-tree                                            
  4       Test          0                            class                                                       
  5         p0          1                         variable  type-tree                                            
  6       main          1  proc-tree             procedure                                                      0
  7          p          2                         variable  type-tree                                            
  8        xxx          2                         variable  type-tree                                            
************* SYNTAX TREE PRINTOUT ***********

  +-[DUMMYnode]
R-[ProgramOp]
  |   +-[STNode,4,"Test"]
  | +-[ClassDefOp]
  | | |         +-[NUMNode,1]
  | | |       +-[AssignOp]
  | | |       | |     +-[DUMMYnode]
  | | |       | |   +-[SelectOp]
  | | |       | |   | | +-[DUMMYnode]
  | | |       | |   | +-[FieldOp]
  | | |       | |   |   +-[STNode,3,"x"]
  | | |       | | +-[VarOp]
  | | |       | | | +-[STNode,7,"p"]
  | | |       | +-[AssignOp]
  | | |       |   +-[DUMMYnode]
  | | |     +-[StmtOp]
  | | |     | |   +-[NUMNode,1]
  | | |     | | +-[AssignOp]
  | | |     | | | |     +-[DUMMYnode]
  | | |     | | | |   +-[SelectOp]
  | | |     | | | |   | | +-[DUMMYnode]
  | | |     | | | |   | +-[FieldOp]
  | | |     | | | |   |   +-[STNode,2,"y"]
  | | |     | | | | +-[VarOp]
  | | |     | | | | | +-[STNode,7,"p"]
  | | |     | | | +-[AssignOp]
  | | |     | | |   +-[DUMMYnode]
  | | |     | +-[StmtOp]
  | | |     |   +-[DUMMYnode]
  | | |   +-[BodyOp]
  | | |   | |       +-[DUMMYnode]
  | | |   | |     +-[CommaOp]
  | | |   | |     | | +-[DUMMYnode]
  | | |   | |     | +-[TypeIdOp]
  | | |   | |     |   +-[INTEGERTNode]
  | | |   | |   +-[CommaOp]
  | | |   | |   | +-[STNode,8,"xxx"]
  | | |   | | +-[DeclOp]
  | | |   | | | +-[DUMMYnode]
  | | |   | +-[BodyOp]
  | | |   |   |       +-[DUMMYnode]
  | | |   |   |     +-[CommaOp]
  | | |   |   |     | | +-[DUMMYnode]
  | | |   |   |     | +-[TypeIdOp]
  | | |   |   |     |   +-[STNode,1,"Point"]
  | | |   |   |   +-[CommaOp]
  | | |   |   |   | +-[STNode,7,"p"]
  | | |   |   | +-[DeclOp]
  | | |   |   | | +-[DUMMYnode]
  | | |   |   +-[BodyOp]
  | | |   |     +-[DUMMYnode]
  | | | +-[MethodOp]
  | | | | |   +-[DUMMYnode]
  | | | | | +-[SpecOp]
  | | | | | | +-[DUMMYnode]
  | | | | +-[HeadOp]
  | | | |   +-[STNode,6,"main"]
  | | +-[BodyOp]
  | |   |       +-[DUMMYnode]
  | |   |     +-[CommaOp]
  | |   |     | | +-[DUMMYnode]
  | |   |     | +-[TypeIdOp]
  | |   |     |   +-[STNode,1,"Point"]
  | |   |   +-[CommaOp]
  | |   |   | +-[STNode,5,"p0"]
  | |   | +-[DeclOp]
  | |   | | +-[DUMMYnode]
  | |   +-[BodyOp]
  | |     +-[DUMMYnode]
  +-[ClassOp]
    |   +-[STNode,1,"Point"]
    | +-[ClassDefOp]
    | | |       +-[DUMMYnode]
    | | |     +-[CommaOp]
    | | |     | | +-[DUMMYnode]
    | | |     | +-[TypeIdOp]
    | | |     |   +-[INTEGERTNode]
    | | |   +-[CommaOp]
    | | |   | +-[STNode,2,"y"]
    | | | +-[DeclOp]
    | | | | |     +-[DUMMYnode]
    | | | | |   +-[CommaOp]
    | | | | |   | | +-[DUMMYnode]
    | | | | |   | +-[TypeIdOp]
    | | | | |   |   +-[INTEGERTNode]
    | | | | | +-[CommaOp]
    | | | | | | +-[STNode,3,"x"]
    | | | | +-[DeclOp]
    | | | |   +-[DUMMYnode]
    | | +-[BodyOp]
    | |   +-[DUMMYnode]
    +-[ClassOp]
      +-[DUMMYnode]
